app/models/community_user.rb

```diff
@@ -152,10 +155,28 @@ def trust_level
     attributes['trust_level'] || recalc_trust_level
   end
 
+  # Checks if the community user is an admin (global or on the current community)
+  # @return [Boolean] check result
+  def admin?
+    is_admin || user&.is_global_admin || false
+  end
+
+  # Checks if the community user is a moderator (global or on the current community)
+  # @return [Boolean] check result
+  def moderator?
+    is_moderator || user&.is_global_moderator || false
```

Comment:
Untested suggestion:

```suggestion
    is_moderator || user&.global_moderator? || false
```

Reply:
Same as with `admin?`, done